src/main/java/org/fogbowcloud/arrebol/execution/DockerTaskExecutor.java

```diff
@@ -1,146 +1,200 @@
         TaskExecutionResult.RESULT result = TaskExecutionResult.RESULT.SUCCESS;
         for (Command cmd : commands) {
             if (cmd.getState().equals(CommandState.FAILED)) {
                 result = TaskExecutionResult.RESULT.FAILURE;
                 break;
             }
         }
-
-        LOGGER.info("Result of task" + task.getId() + ": " + result.toString());
-
         taskExecutionResult = new TaskExecutionResult(result, commandsResults, commands);
         return taskExecutionResult;
     }
 
-    private Integer start() {
+    protected Integer start(){
         try {
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_RUN + " " + this.containerName + " " + this.imageId
-            };
-            Process p = Runtime.getRuntime().exec(cmd);
-
-            Integer exitCode = p.waitFor();
-            return exitCode;
-        } catch (Exception e){
-            throw new RuntimeException("Error while trying execute commands to start container " + this.containerName);
+            LOGGER.info("Starting DockerTaskExecutor " + this.getContainerName());
+            this.workerDockerRequestHelper.start();
+            return new Integer(0);
+        } catch (Exception e) {
+            return new Integer(127);
         }
     }
 
-    private Integer stop() {
+    protected Integer stop(){
         try {
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_STOP + " " + this.containerName
-            };
-            Process p = Runtime.getRuntime().exec(cmd);
-
-            Integer exitCode = p.waitFor();
-            return exitCode;
-
-        } catch (Exception e){
-            throw new RuntimeException("Error while trying execute commands to stop container " + this.containerName);
+            LOGGER.info("Stopping DockerTaskExecutor " + this.getContainerName());
+            this.workerDockerRequestHelper.stop();
+            return new Integer(0);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new Integer(127);
         }
     }
 
-    private Integer executeCommand(Command command) {
-        try {
-            String commandStr = command.getCommand();
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_EXEC + " " + this.containerName + " " + commandStr
-            };
-            ProcessBuilder builder = new ProcessBuilder(cmd);
-            Process process = builder.start();
-            Integer exitCode = process.waitFor();
-            return exitCode;
-
-        } catch(Exception e){
-            throw new RuntimeException("Error while truing execute commands to container " + this.containerName);
+    protected Integer executeCommand(Command command) throws Exception {
+        LOGGER.info("Executing command [" + command.getCommand() + "] in worker [" + this.getContainerName() + "].");
```

Comment:
we need task info

Reply:
Solved in 3f2467d41957697d2578c4d718c4907e36824057.